streams/src/test/java/org/apache/kafka/streams/StreamsBuilderTest.java

```diff
@@ -69,6 +73,30 @@ public void shouldAllowJoinUnmaterializedFilteredKTable() {
         builder.<Bytes, String>stream("stream-topic").join(filteredKTable, MockValueJoiner.TOSTRING_JOINER);
 
         driver.setUp(builder, TestUtils.tempDirectory());
+
+        ProcessorTopology topology = builder.internalTopologyBuilder.build();
+
+        assertThat(topology.stateStores().size(), equalTo(1));
+        assertThat(topology.processorConnectedStateStores("KSTREAM-JOIN-0000000005").size(), is(1));
```

Comment:
This an the one below can be merged into a single assertThat with "equals" and a singelton Set as expected value